adapters/handlers/rest/authz/handlers_authz.go

```diff
@@ -110,30 +110,39 @@ func (h *authZHandlers) createRole(params authz.CreateRoleParams, principal *mod
 }
 
 func (h *authZHandlers) addPermissions(params authz.AddPermissionsParams, principal *models.Principal) middleware.Responder {
-	if *params.Body.Name == "" {
+	if params.ID == "" {
 		return authz.NewAddPermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(errors.New("role name is required")))
 	}
```

Comment:
pretty sure that this cannot happen as ID is now part of the path

Reply:
This check is present in all other endpoints. I think you're right that it may be a needless check but shall we keep it to maintain consistency? Otherwise I'll remove it from the others too, wdyt?

Reply:
> shall we keep it to maintain consistency

I think we need it, when it is in the body but can remove it if it is in the path. Personally don't think we need consistency between endpoints that have it in the body and that have it in the path, but I don't feel strongly about it 🤷

Reply:
No, I mean the check is there for `params.ID` too:
- [get](https://github.com/weaviate/weaviate/blob/stable/v1.28/adapters/handlers/rest/authz/handlers_authz.go#L250)
- [delete](https://github.com/weaviate/weaviate/blob/stable/v1.28/adapters/handlers/rest/authz/handlers_authz.go#L289)
- [assign](https://github.com/weaviate/weaviate/blob/stable/v1.28/adapters/handlers/rest/authz/handlers_authz.go#L316)

Reply:
I will remove these checks from the path easily. You're right that it's impossible for it to be `""` when in the path